validator_client/src/initialized_validators.rs

```diff
@@ -422,10 +455,29 @@ impl InitializedValidators {
     ///
     ///  - The validator is known to `self`.
     ///  - The validator is enabled.
-    pub fn signing_method(&self, voting_public_key: &PublicKeyBytes) -> Option<Arc<SigningMethod>> {
-        self.validators
-            .get(voting_public_key)
-            .map(|v| v.signing_method.clone())
+    pub fn signing_method(
+        &self,
+        voting_public_key: &PublicKeyBytes,
+    ) -> Option<(Arc<SigningMethod>, Option<Client>)> {
+        self.validators.get(voting_public_key).map(|v| {
```

Comment:
An alternative to passing around the hashmap and doing a `get` each time would be to clone the `Client` from the map into each `SigningMethod`. Internally the `Client` [contains an `Arc`](https://docs.rs/reqwest/0.11.11/src/reqwest/async_impl/client.rs.html#59-61), so it's super cheap to clone both in terms of performance and memory usage (it's just a pointer).